File: Typescript/src/findPoisonedDuration.ts
export function findPoisonedDuration(timeSeries: number[], duration: number): number {
	let total = 0;
	for (let i = 1; i < timeSeries.length; i++) {
		total += Math.min(duration, timeSeries[i] - timeSeries[i - 1]);
	}
	return total + duration;
}

File: Dart/lib/findPoisonedDuration.dart
class Solution {
  int findPoisonedDuration(List<int> timeSeries, int duration) {
    int total = duration;
    for (int i = 1; i < timeSeries.length; i++) {
      int diff = timeSeries[i] - timeSeries[i - 1];
      total += (diff < duration) ? diff : duration;
    }
    return total;
  }
}
